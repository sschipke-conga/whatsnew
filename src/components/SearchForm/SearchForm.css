header {
  grid-area: top;
  color: yellow;
}