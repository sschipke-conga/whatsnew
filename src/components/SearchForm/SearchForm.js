import React from 'react';
import './SearchForm.css';

const SearchForm = () => {
  return (
  <header>
    <input />
    <button>Search</button>
  </header>
  )
}

export default SearchForm;