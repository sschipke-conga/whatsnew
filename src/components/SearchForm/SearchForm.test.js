import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  let event = { target: { name: 'search', value: 'test value' } };
  const mockSearchStories= jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper= shallow(<SearchForm 
    searchStories={mockSearchStories}
    />)
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should run searchStories on click of button', () => {
    wrapper.find('button').simulate('click');
    expect(mockSearchStories).toHaveBeenCalled()
  });
  it('should update the state based on the input', () => {
    wrapper.instance().handleChange(event);
    expect(wrapper.state('search')).toEqual('test value')
  });
  it('on handleSubmit it should reset state when button is clicked', () => {
    wrapper.instance().setState({search:'giant babies'});
    expect(wrapper.state('search')).toEqual('giant babies');
    wrapper.find('button').simulate('click')
    expect(wrapper.state('search')).toEqual('')
    expect(mockSearchStories).toHaveBeenCalled()
  })
})