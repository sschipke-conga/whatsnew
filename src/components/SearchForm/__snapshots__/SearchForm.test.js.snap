// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<header>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search articles"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</header>
`;
